internal/service/ch/queries.go

```diff
@@ -45,6 +45,8 @@ const (
 	minGroup       = "min(" + vss.ValueNumberCol + ")"
 	maxGroup       = "max(" + vss.ValueNumberCol + ")"
 	medGroup       = "median(" + vss.ValueNumberCol + ")"
+	firstGroup     = "first_value(" + vss.ValueNumberCol + ")"
+	lastGroup      = "last_value(" + vss.ValueNumberCol + ")"
```

Comment:
Unfortunately, I don't think we want to use these since they are not deterministic, since `first_value` is an alias for `any`
> danger
> The query can be executed in any order and even in a different order each time, so the result of this function is indeterminate. To get a determinate result, you can use the [min](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/min) or [max](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/max) function instead of any.

-- https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/any

> anyLast
>
> Selects the last value encountered. The result is just as indeterminate as for the [any](https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/any) function.

--https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/anylast

Reply:
> As with any, without Window Functions the result will be random if the source stream is not ordered and the return type matches the input type

Maybe the bucket here is an ordered stream. It seems to behave that way.

Reply:
fwiw, this is what I was referencing when initially thinking that the source was ordered https://github.com/DIMO-Network/telemetry-api/blob/main/internal/service/ch/queries.go#L316

Reply:
That’s ordering the buckets.

Reply:
I think it is ordered based on the signal table's create statement, but I don't love relying on that. max and min seem safest unless we hit some performance problems